artbotlib/kerberos.py

```diff
@@ -0,0 +1,37 @@
+import os
+import time
+import subprocess
+
+
+def update_last_kinit_env_var():
+    """
+    This function performs kinit as and when required.
+    The function expects .aws directory in the project base directory.
+    The .aws directory should have a keytab file by name redhat.keytab.
+    :return: None
+    """
+
+    keytab_file = "/tmp/keytab/keytab"
+    kinit_request = subprocess.Popen(["kinit", "-kt", keytab_file, "ocp-build/buildvm.openshift.eng.bos.redhat.com@IPA.REDHAT.COM"],
+                                     stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    output, error = kinit_request.communicate()
+    if error:
+        print(error)
+    os.environ["LAST_KINIT_TIME"] = str(int(time.time()))
```

Comment:
why an env var? just seems unnecessary, you could just set a local var like so:
```suggestion

last_kinit_timestamp: str = ""


def update_last_kinit_var():
    """
    This function performs kinit as and when required.
    The function expects .aws directory in the project base directory.
    The .aws directory should have a keytab file by name redhat.keytab.
    :return: None
    """

    keytab_file = "/tmp/keytab/keytab"
    kinit_request = subprocess.Popen(["kinit", "-kt", keytab_file, "ocp-build/buildvm.openshift.eng.bos.redhat.com@IPA.REDHAT.COM"],
                                     stdout=subprocess.PIPE, stderr=subprocess.PIPE)
    output, error = kinit_request.communicate()
    if error:
        print(error)
    last_kinit_timestamp = str(int(time.time()))
```

Reply:
I changed to always kinit